### PR TITLE
Add VideoPress flow to "tailored signup" flow to skip plugins upsell

### PR DESCRIPTION
**Pull request by jgcaruso**
#### Proposed Changes

* Add a separate "Tailored Signup" helper method to skip plugins upsell

I have not updated all `isNewsletterOrLinkInBioFlow` calls since they don't all apply to hiding the "plugins upsell" page, and since the VideoPress flow is a different flow from the `/create` flows, the flow doesn't work if the VideoPress flowname is included in all of those checks.

@Automattic/shilling if there is a better name for this method I'm happy to make changes. I am just going off the clues I had in the existing code. `skipPluginUpsellFlow` might be a more clear, deliberate name?

#### Testing Instructions

* start the VideoPress signup flow at `/start/videopress`
* Complete the flow and payment
* After payment, you should be sent to the site editor and NOT see the plugins upsell page
* Start a regular signup flow at `/start`, Choose the Premium plan, you should see the plugins upsell after purchase
* Start a link in bio flow at `/setup/intro?flow=link-in-bio`, you should not see the plugins upsell after purchase

#### Pre-merge Checklist

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to # 1283-gh-Automattic/greenhouse

### CODE REVIEW

**Review by roundhill — APPROVED** [review]
Tested great for me ~besides the `link-in-bio` loop we noticed. But seems unrelated as it is in prod as well.~ (Edit: we were using the wrong link)

**Comment by sirbrillig** [comment]
> https://github.com/orgs/Automattic/teams/shilling if there is a better name for this method I'm happy to make changes. I am just going off the clues I had in the existing code. skipPluginUpsellFlow might be a more clear, deliberate name?

Whatever you think makes the most sense is fine.

File: packages/onboarding/src/utils.ts [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
+export const VIDEOPRESS_FLOW = 'videopress';
 
 export const isNewsletterOrLinkInBioFlow = ( flowName: string | null ) => {
 	return Boolean( flowName && [ NEWSLETTER_FLOW, LINK_IN_BIO_FLOW ].includes( flowName ) );
 };
+
+export const isTailoredSignupFlow = ( flowName: string | null ) => {
+	return Boolean(
+		flowName &&
+			( isNewsletterOrLinkInBioFlow( flowName ) || [ VIDEOPRESS_FLOW ].includes( flowName ) )
```
Comment (alshakero) [inline packages/onboarding/src/utils.ts]: Is there a reason for creating an array?

```suggestion
			( isNewsletterOrLinkInBioFlow( flowName ) || VIDEOPRESS_FLOW === flowName )
```

**Review by alshakero — APPROVED** [review]
LGTM!

File: packages/onboarding/src/utils.ts (hunk 1 above)
Comment (jgcaruso) [reply packages/onboarding/src/utils.ts]: That was half me not refactoring all the way, and half me thinking "what if someone wants to add a new item to the list"

But I guess we can let that person change it to an array if they need it.

**Pull request merged**